runtime/runtime/src/actions.rs

```diff
@@ -617,12 +617,30 @@ pub(crate) fn action_deploy_contract(
 }
 
 pub(crate) fn action_deploy_global_contract(
+    account: &mut Account,
     account_id: &AccountId,
+    apply_state: &ApplyState,
     deploy_contract: &DeployGlobalContractAction,
     result: &mut ActionResult,
 ) {
     let _span = tracing::debug_span!(target: "runtime", "action_deploy_global_contract").entered();
 
+    let storage_cost = apply_state
+        .config
+        .fees
+        .storage_usage_config
+        .global_contract_storage_amount_per_byte
+        .saturating_mul(deploy_contract.code.len() as u128);
```

Comment:
hm, I would `checked_mul` here. While `saturating_mul` isn't wrong here as this won't overflow with reasonable inputs (`log2(2^128 / 1E20) ≈ 61` leftover bytes) and there will never be 2^128 gas in the network, at the same time if something does go wrong here, the failure is going to be silent and we had a lot of trouble with that in gas calculations in the past.

Now what sort of handling would be applied here, I'm not sure? Panic is probably not great? fail processing the action early with something like insufficient balance? up to you to decide.

Reply:
I decided to use `saturating_mul` here to avoid panic. Saturation would effectively result in action immediately failing with `ActionErrorKind::LackBalanceForState` because `account.amount().checked_sub(storage_cost)` is `None` in that case, which is kind of "fail processing the action early with something like insufficient balance".